djangosaml2idp/models.py

```diff
@@ -50,6 +50,10 @@ def refresh_metadata(self, force_refresh: bool = False) -> bool:
         ''' If a remote metadata url is set, fetch new metadata if the locally cached one is expired. Returns True if new metadata was set.
             Sets metadata fields on instance, but does not save to db. If force_refresh = True, the metadata will be refreshed regardless of the currently cached version validity timestamp.
         '''
+
+        # TODO: this logic appears to be incorrect.
+        #  In the case local_metadata is Falsy and remote metadata is not set you error on extract
+        #  as it has nothing to extract on.
```

Comment:
@mhindery I added this note. This logic appears to be incorrect. I was adding test coverage for the models and ran across it.

Reply:
Thanks for reporting this. I'll have a look at it, there are some save & validation cases in the admin which were previously causing some issues here so I'll check if it behaves nicely.

Reply:
Yeah if you return it back to it's original state you can see the test fails. It's currently xpasses. That being said it seemed like if that is the correct fix that there are 2 places noted that it seems like we should be returning false.